### ISSUE TITLE
Doesn't work with iTerm2 3.0.0 (stable) or later

### ISSUE THREAD

**Issue by philippetev** [fitztrev/shuttle#181]
Just updated iTerm2 to version 3.0.2 and it still doesn't work. I select the commands, but it appears Shuttle doesn't interact with the new version at all. I guess it's an applescript issue again.

**Comment by thshdw** [comment]
If you have a really old version of shuttle you will have to update your shuttle.json to have the nightly directive ```"iTerm_version":"nightly"```  Also check out this issue. #176 and this pull #175

Also make sure that your global settings reflect the default profile in iTerm or make sure that your iTerm has a default profile and that profile is called "Default" also make sure that your iTerm is not set Don't Open Any Windows in Preferences/General/Startup

I'm using 10.11.5 and I have iTerm 3 nightly everything is working so it seems that its a configuration problem. 

Please check these things and report back.

**Comment by philippetev** [comment]
The issue is about the stable iTerm2 branch, not the nightly one. And my Shuttle version is the latest one.

**Comment by mkoepke** [comment]
Confirmed.   With Shuttle 1.2.6 and iTerm 3.0.4 (latest stable) it does not work.   Shuttle will launch the iTerm2 app, but fails to inject anything on the command line.

Downgraded to iTerm2 2.1.14.    Shuttle works correctly with this version

**Comment by guidoilbaldo** [comment]
Same problem here. Another hint: if iTerm is closed, Shuttle does open it but then the iTerm window crashes and closes all of a sudden, without prompting the "cmd".

**Comment by thshdw** [comment]
Ah! Thanks for the extra info... this is because the shuttle directive ```"iTerm_version":"stable"``` launches the applecripts for the legacy iTerm 2.14 

Even though you are using iTerm stable tell shuttle that you are on the nightly branch using ``` "iTerm_version":"nightly"``` and everything should be okay. 

Now that iTerm 3.x is out and in the Stable branch perhaps shuttle should be more specific with its names. 

Let me know if this works for you.

**Comment by mkoepke** [comment]
Changing it to nightly does correctly launch and inject the command line
with 3.0.4.

Yes you might want to make changes around the iTerm version used.
 Someone has forked and is upgrading the original iTerm program and calling
it iTerm3.    So perhaps something like iTerm_program: iTerm, iTerm2,
iTerm3, (next).   Then iTerm_version based on iTerm_program.   With
iTerm2...  2.x, 3.x, stable, nightly.

**Comment by thshdw** [comment]
@mkoepke Thanks for the update. 

I want shuttle to support iTerm 2.14 as some user may still want to use it. Now that that 3.x is in Stable there will be no apple script changes. 

I propose we have 3 directives 
```
"iTerm_version":"legacy"
```
This supports the legacy 2.14 version 
```
"iTerm_version":"stable"
```
This supports the now stable iTerm 3.x branch 
```
"iTerm_version":"nightly"
```
This supports the nightly versions of iTerm

**Comment by mkoepke** [comment]
That makes sense.

**Comment by thshdw** [comment]
@mkoepke Thanks for the feedback... I'll wait for some more feedback and if users agree I'll put these changes in 1.2.7

**Comment by philippetev** [comment]
No argue from me too. The more options we have, the better!

**Comment by thshdw** [comment]
@philippetev 
Thanks for the feedback. I'll proceed with the following:

```
"iTerm_version":"legacy"
```
This will support the applescripts that work the legacy iTerm 2.14 version 
```
"iTerm_version":"stable"
```
This will support the applescripts that work with the iTerm stable 3.x branch 
```
"iTerm_version":"nightly"
```
This will support the applescripts that work with the nightly versions of iTerm

Until these changes make it to 1.2.7; everyone that is using the new iTerm stable will need to tell shuttle that your using ```nightly```

**Comment by johnpuddephatt** [comment]
Came here with the same problem; telling shuttle I'm using nightly solved it. Thanks 👍

**Comment by thshdw** [comment]
@johnpuddephatt You're welcome! I'll have this fixed in the next version.

**Comment by thshdw** [comment]
Closed! fixes are in 1.2.7
